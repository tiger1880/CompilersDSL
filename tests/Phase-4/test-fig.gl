$ verify how scale and center is being stored

fig kite(scale := 1, center := (2, 2)){

    point p1 := (0, 1)
    point p2 := (0, -1)
    point p3 := (1, 0)
    point p4 := (-1, 0)

}

fig singleStep(scale := 1, center := (0, 0)){

    para p := Para(5, 90, 12) $ single staircase step
}

repeat(int i := 1|i < 10|i++){

    kite(i*i, (0, i))
}

$ creates a staircase
repeat (int i := -10|i >= 0|i++){
    
    singleStep(1, (i, i))
}