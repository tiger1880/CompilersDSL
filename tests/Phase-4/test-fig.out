#include <iostream>
#include <vector>
#include <GL/glut.h>
#include <string>
#include <cstdlib>
#include <cmath>
#include <deque>
#include "standard_lib.hpp"

void kite(double scale = 2, Point *center = new Point(0, 0, 1, 0, 0))
{

    Point *p1 = new Point(0, 1, 2, 0, 0);
    Point *p2 = new Point(0, -1, 2, 0, 0);
    Point *p3 = new Point(1, 0, 2, 0, 0);
    Point *p4 = new Point(-1, 0, 2, 0, 0);
}
void singleStep(double scale = 1, Point *center = new Point(0, 0, 1.0, 0, 0))
{

    Para *p = new Para(5, 90, 12, new Point(0, 0, 1.0, 0, 0), 1);
}

int main(int argc, char **argv)
{

    for (int i = 1; i < 10; i++)
    {

        kite(i * i, new Point(0, i, 1.0, 0, 0));
    }

    for (int i = -10; i >= 0; i++)
    {

        singleStep(1, new Point(i, i, 1.0, 0, 0));
    }

    glutInit(&argc, argv);
    glutInitWindowSize(640, 480);
    glutInitWindowPosition(50, 50);
    glutCreateWindow("Figure");
    glutDisplayFunc(display);
    glutReshapeFunc(reshape);
    init();
    glutMainLoop();

    return 0;
}
